src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -77,12 +77,18 @@
     Optional<Course> findById(long courseId);
 
     @Query("""
-            select c
-            from Course c, User u
+            select distinct c
+            from Course c left join fetch c.exercises, User u
             where u.id = :#{#userId} and c.instructorGroupName member of u.groups
             """)
     List<Course> getCoursesForWhichUserHasInstructorAccess(@Param("userId") Long userId);
 
+    @Query("""
+            select distinct c
+            from Course c left join fetch c.exercises
```

Comment:
```suggestion
            from Course c left join fetch c.exercises e where and TYPE(e) = QuizExercise 
```

I think this would make it even more performant, no? But just a quick suggestion :)

Reply:
Good point, I don't think you can compare the whole Set of exercises directly with the ExerciseType, but I adapted the query a bit so the filtering happens inside the database call. Have a look at 00d0613

Reply:
I just tested it locally and it works with the type operator

Reply:
You are very right :o